### ISSUE TITLE
Code signing fails after integrating Paddle-MAS cocoapod

### ISSUE THREAD

**Issue by jhoughjr** [PaddleHQ/Paddle-MAS#3]
The act of integrating this library prevents the project from code signing.
I was told by support a fix was to disable code signing before integrating and then reenable it.
However, I have tried this 3 times with negative results.

If a dependency can't be code signed, then it can't be used in Mac App Store apps.
I came looking for the pull request I was told was being merged in soon to fix the issue, but I see no such pull request.

**Comment by BrainFeverMedia** [comment]
I've run into the same issue. I also tried running a script for force signing but couldn't get that to work.

```
LOCATION="${BUILT_PRODUCTS_DIR}"/"${FRAMEWORKS_FOLDER_PATH}"
IDENTITY="Mac Distribution: James Grote"
codesign --verbose --force --sign "$IDENTITY" "$LOCATION/Paddle.framework/Versions/current"
```

**Comment by avalore** [comment]
Vendored frameworks aren't currently fully supported by Cocoapods yet on OS X (causes this codesigning issues as you've experienced). We're waiting for this to be supported/fixed, which looks like it's in the pipeline. At the moment we suggest using manual installation. When this is resolved we will let you know :).

**Comment by jhoughjr** [comment]
Manual installation prevents the test target from seeing any framework symbols. Even if it linked with the test target.

There should be a way to manually sign as a post build script, but idk how to get it working.

**Comment by jhoughjr** [comment]
Earlier today I was able to finally get what I needed, a developer id signed app, and a test target that could use Paddle symbols.

I can try to relay what worked here, though it felt sort of hack, so idk how reproducible it is for others.
I started with Xcode 7 beta 3, and a Podfile with 2 targets, one for the app one for the test, and the paddle sdks integrated with cocoa pods.

That gave me a working app, working tests, but no code signing.

I then removed the pod directive for paddle from the app target, but left it for the test target.
I then drug the Paddle.framework into the workspace under the main project(not the pods)
I added a copy files build phase to the app target for the Paddle.framework.

I then set code signing under the app target’s general preferences, and under the build settings for the target.
(setting it under general doesn’t seem to affect what happens at build.This has been my observation with Xcode for a long time.)
I also didn’t choose a wildcard identity but set every option explicitly to the distribution cert I knew was valid.

Next I tried to use the supplied code signing script from the paddle docs, but I couldn’t get it to find my signing identity from the script.
After some googling, I tried a SO suggestion, deleting the code signing script and adding a code signing flag of -deep on the app target.
That worked and gave me the developer id signed app target i needed.
The tests continued to work as they are using the pod integration and didn’t ned code signing.

The interesting thing is that the -deep wasn’t required as warning said it was being ignored.
Another interesting thing is that I tried this before on 6.4 and the previous betas and couldn’t get it to work.
So I hypothesize that perhaps beta 3 has added or fixed something with code signing of embedded libraries in the app target.

I’m not sure why the code signing script wouldn’t work, nor why previous attempts did.
Perhaps the code signing script isn’t needed at all in Xcode 7 > beta 3 but I haven’t tested this as code signing isn’t my idea of a fun time! 
Hopefully this can shed some light on how to come up with a working process until cocoa pods gets a fix.

The tl;dr is you can still use pods to keep the test target happy, but keep them out of the loop as far as paddle and the app target is concerned.
Then try code signing as normal without the run script build phase to explicitly sign the library. It may be a beta 3 only feature.

> On
